src/validation.cpp

```diff
@@ -3907,7 +3907,7 @@ void ChainstateManager::ReceivedBlockTransactions(const CBlock& block, CBlockInd
     // m_chain_tx_count value is not zero, assert that value is actually correct.
     auto prev_tx_sum = [](CBlockIndex& block) { return block.nTx + (block.pprev ? block.pprev->m_chain_tx_count : 0); };
     if (!Assume(pindexNew->m_chain_tx_count == 0 || pindexNew->m_chain_tx_count == prev_tx_sum(*pindexNew) ||
-                pindexNew == CurrentChainstate().SnapshotBase())) {
+                std::any_of(m_chainstates.begin(), m_chainstates.end(), [&](auto& cs) EXCLUSIVE_LOCKS_REQUIRED(cs_main) { return cs->SnapshotBase() == pindexNew; }))) {
```

Comment:
nit f85ff75dad4ce8712fe65a636cf36d57b4066785: use ranges::any_of, and `const auto&`?

Also, in the commit message: `is not just ambiguous because actively confusing` -> `is not just ambiguous but actively confusing`

Reply:
re: https://github.com/bitcoin/bitcoin/pull/30214#discussion_r2596329568

> use ranges::any_of, and `const auto&`?

Make sense, applied change.